components/layout_2020/style_ext.rs

```diff
@@ -634,6 +639,19 @@ impl ComputedValuesExt for ComputedValues {
             return true;
         }
 
+        // From <https://www.w3.org/TR/css-will-change/#valdef-will-change-custom-ident>:
+        // > If any non-initial value of a property would create a stacking context on the element,
+        // > specifying that property in will-change must create a stacking context on the element.
+        if self.clone_will_change().bits.intersects(
+            WillChangeBits::OPACITY |
+                WillChangeBits::PERSPECTIVE |
+                WillChangeBits::STACKING_CONTEXT_UNCONDITIONAL |
+                WillChangeBits::TRANSFORM |
```

Comment:
Perspective and transform should also check `is_transformable()`. Otherwise this is red on Servo:

```html
<!DOCTYPE html>
<style>
body { transform: scale(1); height: 0 }
div { background: green; display: inline; }
div::before { content: ".................."; position: relative; z-index: -1; background: red; }
</style>
<div style="will-change: transform"></div>
<div style="will-change: perspective"></div>
```